lib/ts/recipe/thirdparty/recipeImplementation.ts

```diff
@@ -253,10 +228,28 @@ export default function getRecipeImplementation(recipeId: string, appInfo: Norma
                 })
             );
 
+            if (jsonBody.status === "FIELD_ERROR") {
+                throw new STGeneralError(jsonBody.error);
+            }
+
             return {
                 ...jsonBody,
                 fetchResponse,
             };
         },
+        generateStateToSendToOAuthProvider: function (): string {
+            return generateThirdPartyProviderState();
```

Comment:
Put the logic of `generateThirdPartyProviderState` in here unless it's being used somewhere else..